dumpling/export/status.go

```diff
@@ -72,3 +75,43 @@ func calculateTableCount(m DatabaseTables) int {
 	}
 	return cnt
 }
+
+type StatusRecorder struct {
+	mu             sync.Mutex
+	lastFinished   int64
+	lastUpdateTime time.Time
+	speedBPS       int64
+}
+
+func NewStatusRecorder() *StatusRecorder {
+	return &StatusRecorder{
+		lastUpdateTime: time.Now(),
+	}
+}
+
+func (s *StatusRecorder) getSpeed(finished int64) int64 {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if finished == s.lastFinished {
```

Comment:
```suggestion
	if finished <= s.lastFinished {
```
`finished < s.lastFinished` might not happen, but in case of bugs cause speed negative